### ISSUE TITLE
Feature request: Expand All in the call tree

### ISSUE THREAD

**Issue by agzam** [flow-storm/flow-storm-debugger#106]
When going through deeply nested call tree it gets very tedious to have to expand every node by hand. Would be nice to have "expand all sub-nodes" feature.

**Comment by jpmonettas** [comment]
Yes! I'm not sure if that is straight forward to implement since the tree is lazy. It is retrieving childs from the backend as you expand. This is so we don't load a huge tree when we have millions of records, since probably the user isn't interested in all of them.

If you are interested in the path that leads to a specific function you can jump to the function using the quickbox (or any other means) and then hit the `show the current frame` button on the tree tool. It should expand all nodes leading to that function.

**Comment by agzam** [comment]
Hmm.. maybe then expanding just a limited number of levels ¯\\\_(ツ)_/¯

> interested in the path that leads to a specific function

That's the thing, I'm not sure which function down the nested call tree is failing until I expand every level.

Also, somewhat related. I am currently having to spend a lot of time digging through the call tree and what's slightly annoying is that most of the time my highlighted row ends up being at the bottom of the window. So it becomes like a tedious thing - expand & scroll up, expand & scroll up.

I wish for something like Emacs's `(recenter)` - `C-l` command. Also, maybe some margin at the bottom, so you can scroll past the last line, otherwise when you have multiple nested levels at the bottom, you can't bring it to the center of the window.

**Comment by jpmonettas** [comment]
> That's the thing, I'm not sure which function down the nested call tree is failing until I expand every level.

You mean failing with an exception? If that is the case you can eval `:ex` on the repl to make the debugger jump to the last captured exception. Or you can open the thread and step to the last step.

If it is not related to an exception I'm not sure how does it help to have all the nodes expanded if the only strategy is clicking every sub node.

I think I'm missing something. Can you expand a little bit on  the situation?

> Also, somewhat related. I am currently having to spend a lot of time digging through the call tree and what's slightly annoying is that most of the time my highlighted row ends up being at the bottom of the window. So it becomes like a tedious thing - expand & scroll up, expand & scroll up.

yeah, that is very annoying, but I haven't figured that out yet, the openjfx TreeView is jumping like that.

> I wish for something like Emacs's (recenter) - C-l command. Also, maybe some margin at the bottom, so you can scroll past the last line, otherwise when you have multiple nested levels at the bottom, you can't bring it to the center of the window without expanding the sub-levels. And if you have everything expanded, you still can't scroll up past the last row. From UX standpoint that's quite inconvenient.

All this is also probably good ideas, but again, the default TreeView isn't very flexible, maybe it is possible by extending the jfx TreeView.

 > Additionally, would be cool if it was possible to navigate the tree without arrow keys - C-n/C-p & C-f/C-b (for vanilla Emacs folk) or hjkl (for Vim/Evil lovers)

this should probably also needs a TreeView extension.

All probably doable, just not enough time, since those kind of extensions are pretty time consuming, that is why I haven't tried them yet.
